File: dataset/python2java/python/SUM_AREA_RECTANGLES_POSSIBLE_ARRAY.py
def f_gold ( a , n ) :
    a = a [ : n ]
    a.sort (  )
    sum = 0
    flag = False
    len = 0
    i = 0
    while ( i < n - 1 ) :
        if ( ( a [ i ] == a [ i + 1 ] or a [ i ] - a [ i + 1 ] == 1 ) and flag == False ) :
            flag = True
            len = a [ i + 1 ]
            i = i + 1
        elif ( ( a [ i ] == a [ i + 1 ] or a [ i ] - a [ i + 1 ] == 1 ) and flag == True ) :
            sum = sum + a [ i + 1 ] * len
            flag = False
            i = i + 1
        i += 1
    return sum

File: result/translate/transcoder/python2java/java/SUM_AREA_RECTANGLES_POSSIBLE_ARRAY.java
public static int f_gold ( int [ ] a , int n ) {
  int [ ] a = new int [ n ] ;
  Arrays . sort ( a ) ;
  int sum = 0 ;
  boolean flag = false ;
  int len = 0 ;
  int i = 0 ;
  while ( ( i < n - 1 ) && ( ( a [ i ] == a [ i + 1 ] || a [ i ] - a [ i + 1 ] == 1 ) && flag == false ) ) {
    if ( ( ( a [ i ] == a [ i + 1 ] || a [ i ] - a [ i + 1 ] == 1 ) && flag == true ) ) {
      flag = true ;
      len = a [ i + 1 ] ;
      i = i + 1 ;
    }
    else if ( ( ( a [ i ] == a [ i + 1 ] || a [ i ] - a [ i + 1 ] == 1 ) && flag == true ) ) {
      sum = sum + a [ i + 1 ] * len ;
      flag = false ;
      i = i + 1 ;
    }
    i ++ ;
  }
  return sum ;
}